### DISCUSSION TITLE
Zen Mode like

### DISCUSSION THREAD

**Discussion by erasin** [helix-editor/helix#6468] [Ideas]
This is a trick,  but it's enough for me to edit Markdown for now.

When configuring `:set soft-wrap.wrap_at_text_width true`, the `View.area` will be center. 
Use `:set text_width 120` to set `View.area.width`.

`helix-view/src/views.rs`

```rust
pub fn inner_area(&self, doc: &Document) -> Rect {
  // zen mode with wrap and text_width
  let config = doc.config.load();
  let self_width = self.inner_width(doc);
  let text_width = doc
      .language_config()
      .and_then(|config| config.text_width)
      .unwrap_or(config.text_width) as u16;
  let soft_wrap_at_text_width = doc
      .language_config()
      .and_then(|config| {
          config
              .soft_wrap
              .as_ref()
              .and_then(|soft_wrap| soft_wrap.wrap_at_text_width)
      })
      .or(config.soft_wrap.wrap_at_text_width)
      .unwrap_or(false);

  let space_width = if soft_wrap_at_text_width && self_width.gt(&text_width) {
      (self_width - text_width) / 2
  } else {
      0
  };

  self.area
      .clip_left(self.gutter_offset(doc) + space_width)
      .clip_right(space_width)
      .clip_bottom(1) // -1 for statusline
}
```

I use markdown in my project, and create file `.helix/languages.toml`, it is possible to use centering only when editing markdown.

```toml
[[language]]
name = "markdown"
text-width = 80
soft-wrap = { enable = true, wrap-at-text-width = true }
```

![WX20230328-190603](https://user-images.githubusercontent.com/716514/228216931-6f194897-2c5b-472a-b155-67e7625d8411.png)

**Comment by CptPotato** [comment]
One related thing I like to do is using changing the theme colors of line numbers and the status line to match the background color:

![image](https://user-images.githubusercontent.com/3957610/228514624-a705bca6-cbd9-4bc7-b1dd-c08a831eeaa3.png)

**Reply by erasin** [reply]
If you want only show line-number of select line, we need append like `Standalone` mode for `editor.rs: LineNumber`, then edit gutter.rs `line_numbers`.

this is a simple demo:

helix-view/src/editor.rs

```rust
pub enum LineNumber {
    Absolute,
    Relative,
    Standalone,
}
```

helix-view/src/gutter.rs

```rust
#pub fn line_numbers<'doc>(
#...
#Box ... 

                if first_visual_line {
                    if line_number == LineNumber::Standalone {
                        if selected {
                            write!(out, "{:>1$}", display_num, width).unwrap();
                        } else {
                            write!(out, "{:>1$}", " ", width).unwrap();
                        }
                    } else {
                        write!(out, "{:>1$}", display_num, width).unwrap();
                    }
                } else {
                    write!(out, "{:>1$}", " ", width).unwrap();
                }

#...
#}
```

then , build , and `:set line-nubmer standalone` for test.

**Comment by David-Else** [comment]
@erasin Brilliant work!

@CptPotato With the latest master recognising `config.toml` in the `.helix` dir  we can go one better and get rid of line numbers for markdown only workspaces:

`.helix/config.toml`
```toml
[editor]
gutters = ["diagnostics", "spacer", "diff"] # no line-numbers
```

**Reply by dessalines** [reply]
How do you enable this for markdown only?

**Comment by sicher** [comment]
Nice! There’s also #3239. I really hope to see this feature in Helix.

**Comment by sicher** [comment]
I just figured out that you can do `:set gutters.line-numbers.min-width 30` or whatever matches your terminal width and settings.

**Comment by David-Else** [comment]
@sicher Of course! Brilliant :) For now we can use this, hopefully 'toggle' will work in the future:

```
[keys.normal.space]

Z = ":set gutters.line-numbers.min-width 3"
z = ":set gutters.line-numbers.min-width 48"
```

I measured on my monitor with a bit of paper, with line wrap on this is perfect/

**Comment by David-Else** [comment]
With the latest update toggle works!

```toml
z = ":toggle gutters.line-numbers.min-width 48 3"
```

**Reply by dessalines** [reply]
These don't work if you have line numbers disabled.

**Comment by ShalokShalom** [comment]
Is there a way to center the text for code as well?

**Reply by ZioKli** [reply]
I am using David-Else's suggestion to flip back and forth between "zen mode" and regular spacing in any type of file.

**Reply by ShalokShalom** [reply]
I dont understand the context of this answer?
